warehouse/query-core/src/main/java/datawave/query/index/lookup/Union.java

```diff
@@ -311,4 +310,87 @@ public String getContextDebug() {
     public JexlNode currentNode() {
         return currNode;
     }
+    
+    /**
+     * Seek every IndexStream whose top shard is lower than the specified shard.
+     *
+     * @param seekShard
+     *            the shard to seek to.
+     * @return the lowest shard within this union after seeking
+     */
+    @Override
+    public String seek(String seekShard) {
+        
+        // Guard against the case when a seek range like YYYYMMDD_ is provided.
+        if (seekShard.charAt(seekShard.length() - 1) == '_')
+            seekShard = new String(seekShard.getBytes(), 0, seekShard.length() - 1);
+        
+        // Check the current element first
+        String topShard = isTopElementAMatch(seekShard);
+        if (topShard != null) {
+            // If the top element is a day and we are seeking to a shard range within the day, re-map the top element to the shard range.
+            // This allows us to actually intersect.
```

Comment:
We are in a Union here, do we really want to intersect?  If we have a day range, then why would we start returning shard ranges?  Would we then potentially miss some shards within the day?

Reply:
This is an artifact of how the Intersection sorts the index streams by top shard. 

Take `(A_shard && (B_day || C_day))`  for example. The union's top shard is a day, but must be mapped to the same shard as the A term in order for the intersection to recognize that an intersection is possible.

Reply:
perhaps add to the comment what you just said here for clarity.

Reply:
Will do